packages/next/client/link.tsx

```diff
@@ -268,6 +277,8 @@ const Link = React.forwardRef<HTMLAnchorElement, LinkPropsReal>(
     }
 
     const p = prefetchProp !== false
+    // @ts-ignore useTransition exists
+    const [, /* isPending */ startTransition] = React.useTransition()
```

Comment:
https://github.com/vercel/next.js/pull/37332/files#r886331739

Reply:
Added an extra check to this one to ensure it's not called with React 17, good catch 👍